### PR TITLE
update versions map.

### PR DESCRIPTION
**Pull request by morvencao**

### CODE REVIEW

File: version/versions.yaml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 - operatorVersion: 1.3.0
   supportedIstioVersions: 1.3.0
   recommendedIstioVersions: 1.3.0
```
Comment (ostromart) [inline version/versions.yaml]: Since we're updating this, could you please add an entry for 1.3.1 too?

File: version/versions.yaml [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
-
+- operatorVersion: 1.4.0
+  supportedIstioVersions: ">=1.3.0,<=1.4"
```
Comment (ostromart) [inline version/versions.yaml]: We haven't yet finalized whether we'll have forward compatibility in a minor version, so let's keep this to 1.4.0 for now.

File: version/versions.yaml (hunk 1 above)
Comment (morvencao) [reply version/versions.yaml]: done

File: version/versions.yaml (hunk 2 above)
Comment (morvencao) [reply version/versions.yaml]: removed 1.4.0 entry.

**Review by ostromart — APPROVED** [review]

**Pull request merged**